### DISCUSSION TITLE
Ayuda instalando con mac M1

### DISCUSSION THREAD

**Discussion by mt-garcia** [IIC2133-PUC/2024-1#3] [Ayuda] [answered]
Hola!!
Estoy media perdida con las instalaciones y quizás lo hice mal.
Instalé docker desktop y después salía que había que clonar la repo, pero creo que cloné otra porque hice esta:
https://github.com/IIC2133-PUC/edd-docker

Porque después encontré esta: https://github.com/IIC2133-PUC/edd-runner
Pero, con esa no logré hacer el "make install" porque me decía que no tenía los permisos.

Entonces no entendí por qué habían dos cosas, cuál me sirve para qué y cómo hacerlo para ocuparlo. Se hace directo en la terminal? o por qué se instala el docker desktop? osea, entendí la razón, no sé cómo usarlo.

Gracias!

Esto me pasó además intentando clonar el repo
<img width="640" alt="Captura de Pantalla 2024-03-08 a la(s) 20 36 46" src="https://github.com/IIC2133-PUC/2024-1/assets/69651156/15a0349b-e7f2-4593-9490-6e537ca1afd2">

**Comment by CarloGauss33** [comment, marked as answer]
Hola @mt-garcia Sobre el primer punto de edd-docker. ejecuta con `sudo make install` de esa forma se le asignaran todos los permisos
Sobre el segundo, se ve extraño, según encontre puede deberse a la conexión de internet. En otra red te funciona? Si no intenta con 

`git clone --depth=1  https://github.com/IIC2133-PUC/2024-1` 

Este comando solo clonara los contenidos sin la historia por lo que deberia ser menos pesada la ejecución del comando

**Reply by mt-garcia** [reply]
Ahí me funcionó todo!! parece que sí era el internet también, gracias!!
Ahora eso sí estoy media perdida perdida, porque no sé si tengo que programar en docker o en el programa que uso normalmente, o en la terminal, o cómo funciona docker. Y están los talleres y además el link de los apuntes del dcc y no supe por dónde partir, me confundí entera :(

**Reply by sachondo7** [reply]
Hola! Docker es una app que en este curso sirve para verificar que no tengas leaks de memoria. 
Mi recomendación es que hagas tu tarea en tu ambiente que utilices normalmente (sea vscode o el que te guste), y una vez la hayas terminado, copies la carpeta que contiene tu codigo a docker y lo ejecutes ahí para verificar lo de los leaks. (esto lo enseñan en las capsulas)
No recomiendo hacer la tarea directamente en docker, ya que el tiempo de ejecución suele ser mucho más lento. 
Saludos!
